### PR TITLE
Feature/giant squid banner

### PR DESCRIPTION
**Pull request by osangel**
## Description

Add giant-squid-banner
UI:https://www.figma.com/file/IlNFZbcudqcF8qjugLPP85/Giant-Squid?node-id=187%3A2&t=Q4syPi8kKDOysZxO-0
image

---

Before we can merge this PR, please make sure that all the following items have been checked off. If any of the checklist items are not applicable, please leave them but write a little note why.

- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work
- [ ] Re-reviewed files changed in the Github PR explorer

## If PR to `staging`
- [ ] Updated relevant documentation in the code
- [ ] Test code changes manually, and describe your procedure in a comment on this PR
- [ ] Mark this PR with the correct milestone

## If PR to `main`
- [ ] Update the version number in `package.json` and `src/config/common.json`
- [ ] Update the minimum required signer version number in `src/config/common.json` if applicable
- [ ] Test following the procedure in `docs/manual_test.md`
- [ ] Update `CHANGELOG.md`
- [ ] Make sure that all PR's to `manta-signer` and `sdk` on which this PR depends are merged

### CODE REVIEW

File: src/components/ActivityBanner/GiantSquidBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  return (
+    <div
+      className={
+        'flex h-68 cursor-pointer items-center justify-center bg-giant-squid font-red-hat-mono text-sm font-semibold leading-19 text-banner'
+      }
+      onClick={() =>
+        window.open('https://galxe.com/calamarinetwork/campaigns')
+      }>
+      <div className="mr-4">
+        KMA holders can participate with the Giant Squid Program on Galxe.com
```
Comment (Kevingislason) [inline src/components/ActivityBanner/GiantSquidBanner.tsx:14]: I would say "KMA holders can participate in the Giant Squid Program on Galxe.com"

File: src/components/ActivityBanner/index.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React, { useEffect, useState } from 'react';
+import { giantSquidTimeDuring } from 'utils/time/timeDuring';
+import GiantSquidBanner from './GiantSquidBanner';
+export enum BANNER_TYPE {
+  GIANT_SQUID = 'GIANT_SQUID',
+  DEFAULT = 'DEFAULT'
+}
```
Comment (Kevingislason) [inline src/components/ActivityBanner/index.tsx:7]: You don't need these string values, you can just do this:
```js
export enum BANNER_TYPE {
  GIANT_SQUID,
  DEFAULT,
}

```

File: src/components/ActivityBanner/index.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+const ActivityBanner: React.FC<object> = () => {
+  const [bannerType, setBannerType] = useState(BANNER_TYPE.DEFAULT);
+
+  useEffect(() => {
+    if (giantSquidTimeDuring()) {
+      setBannerType(BANNER_TYPE.GIANT_SQUID);
+    } else {
+      setBannerType(BANNER_TYPE.DEFAULT);
+    }
+  }, []);
```
Comment (Kevingislason) [inline src/components/ActivityBanner/index.tsx:17]: nit: it would be easier to read this file if you added a newline after this useEffect

File: src/resources/icons/index.ts [hunk 4]
```diff
@@ -81,6 +81,9 @@ import { ReactComponent as txSuccess } from 'resources/icons/tx-success.svg';
 import { ReactComponent as txFailed } from 'resources/icons/tx-failed.svg';
 import { ReactComponent as txPending } from 'resources/icons/tx-pending.svg';
 
+//avtivity banner icons
```
Comment (Kevingislason) [inline src/resources/icons/index.ts:84]: nit: I think you can remove this comment, we usually don't comment imports, and this import is self-explanatory

File: src/utils/time/timeDuring.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  return judgeDuringTime(
+    new Date().toUTCString(),
+    new Date(giantSquidStartTimeStr).toUTCString(),
+    new Date(giantSquidEndTimeStr).toUTCString()
+  );
+};
+export const judgeDuringTime = (
+  timeStr: string,
+  startTimeStr: string,
+  endTimeStr: string
```
Comment (Kevingislason) [inline src/utils/time/timeDuring.tsx:15]: I think it makes more sense to accept `Date`s as parameters rather than strings

File: src/utils/time/timeDuring.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+const giantSquidEndTimeStr = 'Fri Mar 31 2023 00:00:00 GMT+0800 (中国标准时间)';
+
+export const giantSquidTimeDuring = (): boolean => {
+  return judgeDuringTime(
+    new Date().toUTCString(),
+    new Date(giantSquidStartTimeStr).toUTCString(),
+    new Date(giantSquidEndTimeStr).toUTCString()
+  );
+};
+export const judgeDuringTime = (
```
Comment (Kevingislason) [inline src/utils/time/timeDuring.tsx:12]: I find these function names a bit hard to parse. I would name them `getGiantSquidBannerIsActive` and `getTimeWindowIsActive`

File: src/components/ActivityBanner/index.tsx (hunk 3 above)
Comment (osangel) [reply src/components/ActivityBanner/index.tsx:17]: Done

**Review by Kevingislason — APPROVED** [review]
